### DISCUSSION TITLE
Does pulsar delayed index use Heap Memory or Direct Memory？

### DISCUSSION THREAD

**Discussion by tianshimoyi** [apache/pulsar#19916] [Q&A] [answered]
pulsar version:  2.9.4

**Comment by mattisonchao** [comment, marked as answer]
Direct. : ) 

Link: https://github.com/apache/pulsar/blob/master/pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/TripleLongPriorityQueue.java
